src/weapons.cpp

```diff
@@ -132,6 +135,19 @@ bool Weapons::registerEvent(Event_ptr event, const pugi::xml_node&)
 	return result.second;
 }
 
+bool Weapons::registerLuaEvent(Weapon* event)
+{
+	Event_ptr w{ event };
+	Weapon* weapon = static_cast<Weapon*>(w.release()); //event is guaranteed to be a Weapon
+
+	if (weapons.find(weapon->getID()) != weapons.end()) {
+		weapons[weapon->getID()] = weapon;
+	} else {
+		weapons.emplace(weapon->getID(), weapon);
+	}
```

Comment:
```c++
weapons[weapon->getID()] = weapon;
```
Will insert, or assign, so you don't have to do a `find` lookup.

Reply:
I left it like that so we could easily add a `cout` if people find it necessary to print if a weapon gets overwritten but yea I get your point.
If no one wants the `cout` option then I'll change it to what you suggested.

Reply:
put your suggestion into place in [9148138](https://github.com/otland/forgottenserver/pull/2571/commits/91481384539c767c1ab0570a52f8c011a022bb1b)